src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -268,52 +286,288 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
+        {
+            var symbol = BindPropertyPatternMember(inputType, name, ref hasErrors, diagnostics);
+
+            if (inputType.IsErrorType() || hasErrors)
+            {
+                propertyType = CreateErrorType();
+                return null;
             }
+
+            propertyType = symbol.GetTypeOrReturnType();
+            return symbol;
         }
 
+        private Symbol BindPropertyPatternMember(
+            TypeSymbol inputType,
+            IdentifierNameSyntax memberName,
+            ref bool hasErrors,
+            DiagnosticBag diagnostics)
+        {
+            // TODO: consider refactoring out common code with BindObjectInitializerMember
+            BoundImplicitReceiver implicitReceiver = new BoundImplicitReceiver(memberName, inputType);
+            string name = memberName.Identifier.ValueText;
+
+            BoundExpression boundMember = BindInstanceMemberAccess(
+                node: memberName,
+                right: memberName,
+                boundLeft: implicitReceiver,
+                rightName: name,
+                rightArity: 0,
+                typeArgumentsSyntax: default(SeparatedSyntaxList<TypeSyntax>),
+                typeArguments: default(ImmutableArray<TypeSymbol>),
+                invoked: false,
+                diagnostics: diagnostics);
+
+            if (boundMember.Kind == BoundKind.PropertyGroup)
+            {
+                boundMember = BindIndexedPropertyAccess(
+                    (BoundPropertyGroup)boundMember, mustHaveAllOptionalParameters: true, diagnostics: diagnostics);
+            }
+
+            LookupResultKind resultKind = boundMember.ResultKind;
+            hasErrors |= boundMember.HasAnyErrors || implicitReceiver.HasAnyErrors;
+
+            switch (boundMember.Kind)
+            {
+                case BoundKind.FieldAccess:
+                case BoundKind.PropertyAccess:
+                    break;
+
+                case BoundKind.IndexerAccess:
+                case BoundKind.DynamicIndexerAccess:
+                case BoundKind.EventAccess:
+                    // PROTOTYPE(patterns2): we need to decide what kinds of members can be used in a property pattern.
+                    // For now we support fields and readable non-indexed properties.
+                default:
+                    if (!hasErrors)
+                    {
+                        switch (boundMember.ResultKind)
+                        {
+                            case LookupResultKind.Empty:
+                                Error(diagnostics, ErrorCode.ERR_NoSuchMember, memberName, implicitReceiver.Type, name);
+                                break;
+
+                            case LookupResultKind.Inaccessible:
+                                boundMember = CheckValue(boundMember, BindValueKind.RValue, diagnostics);
```

Comment:
>boundMember = CheckValue(boundMember, BindValueKind.RValue, diagnostics); [](start = 32, length = 73)

This side effect (the assignment to boundMember) probably shouldn't be conditioned on ```hasErrors```. If the only reason we call ```CheckValue``` is to report an error, we can leave ```boundMember``` unchanged.

Reply:
This formulation maximizes the carrying forward of error flags. If you have a specific scenario that you think would be improved by a change here, I'd be happy to look at it.

---
In reply to: [156520936](https://github.com/dotnet/roslyn/pull/23209#discussion_r156520936) [](ancestors = 156520936)